inc/zoo/swar/associative_iteration.h

```diff
@@ -497,11 +497,11 @@ wideningMultiplication(SWAR<NB, T> multiplicand, SWAR<NB, T> multiplier) {
    constexpr auto
        HalfLane = S::NBits,
        UpperHalfOfLanes = SWAR<S::NBits, T>::oddLaneMask().value();
-   auto [res_even, res_odd] = doublingMultiplication(multiplicand, multiplier);
-   auto result = halvePrecision(res_even, res_odd);
+   auto [lower, upper] = doublingMultiplication(multiplicand, multiplier);
+   auto result = halvePrecision(lower, upper);
    auto
-       over_even = D{(res_even.value() & UpperHalfOfLanes) >> HalfLane},
-       over_odd = D{(res_odd.value() & UpperHalfOfLanes) >> HalfLane};
+       over_even = D{(lower.value() & UpperHalfOfLanes) >> HalfLane},
+       over_odd = D{(upper.value() & UpperHalfOfLanes) >> HalfLane};
```

Comment:
shift intra lane allows you to provide the mask.
Please use those primitives instead of deploying the pick-axe